docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+1000×2 DataFrame
+  Row │ Age    Age2
+      │ Int64  Int64
+──────┼──────────────
+    1 │    67   4489
+    2 │    22    484
+    3 │    49   2401
+    4 │    45   2025
+    5 │    53   2809
+    6 │    35   1225
+    7 │    53   2809
+    8 │    35   1225
+  ⋮   │   ⋮      ⋮
+  994 │    30    900
+  995 │    50   2500
+  996 │    31    961
+  997 │    40   1600
+  998 │    38   1444
+  999 │    23    529
+ 1000 │    27    729
+     985 rows omitted
+```
+
+In the example above we returned a `NamedTuple`. Note, however, that returning a `NamedTuple` is not the same as 
+returning a vector of `NamedTuples` as this is allowed.
+
+Let us discuss some other examples using `select`. Often we want to apply some function not to the whole
+column of a data frame, but rather to its individual elements. Normally we can achieve this using broadcasting
+like this:
+
+```jldoctest dataframe
+julia> select(german, :Sex => (x -> uppercase.(x)) => :Sex)
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ MALE
+    2 │ FEMALE
+    3 │ MALE
+    4 │ MALE
+    5 │ MALE
+    6 │ MALE
+    7 │ MALE
+    8 │ MALE
+  ⋮   │   ⋮
+  994 │ MALE
+  995 │ MALE
+  996 │ FEMALE
+  997 │ MALE
+  998 │ MALE
+  999 │ MALE
+ 1000 │ MALE
+985 rows omitted
+```
+
+This pattern is encountered very often in practice, therefore there is a `ByRow` convenience wrapper for a 
+function that creates its broadcasted variant. In these examples `ByRow` type is a special type used for 
+selection operations to signal that the wrapped function should be applied to each element (row) of the selection. 
+Here we are passing `ByRow` wrapper to target column name `:Sex` using `uppercase` function:
+
+```jldoctest dataframe
+julia> select(german, :Sex => ByRow(uppercase) => :SEX) 
+1000×1 DataFrame
+  Row │ SEX
+      │ String
+──────┼────────
+    1 │ MALE
+    2 │ FEMALE
+    3 │ MALE
+    4 │ MALE
+    5 │ MALE
+    6 │ MALE
+    7 │ MALE
+    8 │ MALE
+  ⋮   │   ⋮
+  994 │ MALE
+  995 │ MALE
+  996 │ FEMALE
+  997 │ MALE
+  998 │ MALE
+  999 │ MALE
+ 1000 │ MALE
+985 rows omitted
+```
+
+In this case we transformed our source column `:Age` using `ByRow` wrapper:
+
+```jldoctest dataframe
+julia> select(german, :Age, :Age => ByRow(sqrt)) 
+1000×2 DataFrame
+  Row │ Age    Age_sqrt
+      │ Int64  Float64
+──────┼─────────────────
+    1 │    67   8.18535
+    2 │    22   4.69042
+    3 │    49   7.0
+    4 │    45   6.7082
+    5 │    53   7.28011
+    6 │    35   5.91608
+    7 │    53   7.28011
+    8 │    35   5.91608
+  ⋮   │   ⋮       ⋮
+  994 │    30   5.47723
+  995 │    50   7.07107
+  996 │    31   5.56776
+  997 │    40   6.32456
+  998 │    38   6.16441
+  999 │    23   4.79583
+ 1000 │    27   5.19615
+        985 rows omitted
+```
+
+We can skip specifying a target column name, in which case it is generated automatically by suffixing source
+column name by function name that is applied to it. For example:
```

Comment:
you have given almost identical example above. Why duplicating?

Reply:
In the above example, I was providing `source_column_name`, `transformation`, and `target_column_name` but here only `source_column_name` and `transformation`.

Reply:
It seems to me that in line https://github.com/JuliaData/DataFrames.jl/pull/2717/files#diff-80725fcadb5509a5fb0986e534b25ae5c65bab29c2903e8319c67c7a42143ffaR1734 above you do:
```
select(german, :Age, :Age => ByRow(sqrt)) 
```
without specifying target column name